xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp

```diff
@@ -348,10 +348,28 @@ void CAESinkPipewire::EnumerateDevicesEx(AEDeviceInfoList& list, bool force)
                                   streamTypes.end());
     }
 
-    if (device.m_channels.Count() == 2 && !device.m_streamTypes.empty())
+    // If DTS-HD-MA or TrueHD are configured 8 channels are needed
+    bool hasHBRFormat = std::find_if(device.m_streamTypes.cbegin(), device.m_streamTypes.cend(),
+                                     [](const auto& streamType)
+                                     {
+                                       if (streamType == CAEStreamInfo::STREAM_TYPE_TRUEHD ||
+                                           streamType == CAEStreamInfo::STREAM_TYPE_DTSHD_MA)
+                                         return true;
+
+                                       return false;
+                                     }) != device.m_streamTypes.cend();
```

Comment:
The correct STL algorithm to use here would be `std::any_of`. Also the lambda can be simplified:
```suggestion
    bool hasHBRFormat = std::any_of(device.m_streamTypes.cbegin(), device.m_streamTypes.cend(),
                                    [](const auto& streamType)
                                    {
                                      return streamType == CAEStreamInfo::STREAM_TYPE_TRUEHD ||
                                             streamType == CAEStreamInfo::STREAM_TYPE_DTSHD_MA;
                                    });
```